module-apps/apps-common/widgets/TimeFixedWidget.cpp

```diff
@@ -36,125 +37,152 @@ namespace gui
-
-        hBox->resizeItems();
-    }
+        mainBox = new gui::HBox(this, 0, 0, w, h);
+        mainBox->setAlignment(gui::Alignment(gui::Alignment::Horizontal::Center, gui::Alignment::Vertical::Center));
+        mainBox->setEdges(gui::RectangleEdge::None);
+        mainBox->activeItem = false;
 
-    void TimeFixedWidget::setFirst(const uint32_t first)
-    {
-        const auto decimal = first / 10;
-        if (decimal != 0u) {
-            const auto widestDigit = getWidestDigitWidth(firstPrimoText->getTextFormat().getFont());
-            firstPrimoText->setSize(widestDigit, hBox->widgetArea.h);
-            firstPrimoText->setText(std::to_string(first / 10));
-            firstSecundoText->setText(std::to_string(first % 10));
+        leftBox.box = new gui::HBox(mainBox, 0, 0, 0, 0);
+        leftBox.box->setAlignment(gui::Alignment(gui::Alignment::Horizontal::Right, gui::Alignment::Vertical::Center));
+        leftBox.box->setEdges(gui::RectangleEdge::None);
+
+        attachLabelToBox(colon, mainBox);
+        setColon();
+
+        rightBox.box = new gui::HBox(mainBox, 0, 0, 0, 0);
+        rightBox.box->setAlignment(gui::Alignment(gui::Alignment::Horizontal::Left, gui::Alignment::Vertical::Center));
+        rightBox.box->setEdges(gui::RectangleEdge::None);
+
+        attachLabelToBox(leftBox.minus, leftBox.box);
+        setMinus();
+
+        for (auto &digit : leftBox.digits) {
+            attachLabelToBox(digit, leftBox.box);
         }
-        else {
-            firstPrimoText->setText("");
-            firstPrimoText->setSize(0, 0);
-            firstSecundoText->setText(std::to_string(first));
+
+        for (auto &digit : rightBox.digits) {
+            attachLabelToBox(digit, rightBox.box);
         }
-        firstHBox->resizeItems();
+
+        mainBox->resizeItems();
     }
 
-    void TimeFixedWidget::setSecond(const uint32_t second)
+    void TimeFixedWidget::attachLabelToBox(Label *&label, HBox *&box) const
     {
-        const auto decimal = second / 10;
-        if (decimal != 0u) {
-            secondPrimoText->setText(std::to_string(second / 10));
-            secondSecundoText->setText(std::to_string(second % 10));
-        }
-        else {
-            secondPrimoText->setText("0");
-            secondSecundoText->setText(std::to_string(second));
+        label = new gui::Label(box, 0, 0, 0, 0);
+        label->setEdges(gui::RectangleEdge::None);
+        label->setAlignment(gui::Alignment(gui::Alignment::Horizontal::Center, gui::Alignment::Vertical::Center));
+    }
+
+    void TimeFixedWidget::setMinus() const
+    {
+        leftBox.minus->setText(minusVisible ? minusSign : "");
+    }
+
+    void TimeFixedWidget::setColon() const
+    {
+        colon->setText(colonSign);
+        colon->activeItem = false;
+    }
+
+    void TimeFixedWidget::setLeftBox(uint32_t minutes) const
+    {
+        bool isLimitExceed = minutes > 999;
+        minutes            = isLimitExceed ? minutes % 1000 : minutes;
```

Comment:
```suggestion
        minutes            = minutes % 1000;
```

Reply:
I did change to
  auto rangeGuard = 1000;
  minutes %= rangeGuard;

Reply:
I did change to
  auto rangeGuard = 1000;
  minutes %= rangeGuard;